src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyCacheMaxSizeAction.java

```diff
@@ -0,0 +1,182 @@
+// TODO: Split the cache action into separate actions for different caches.
+
+public class ModifyCacheMaxSizeAction implements Action {
+  private static final Logger LOG = LogManager.getLogger(ModifyCacheMaxSizeAction.class);
+  public static final String NAME = "modifyCacheCapacity";
+  public static final long COOL_OFF_PERIOD_IN_MILLIS = 300 * 1_000;
+
+  private final NodeKey esNode;
+  private final ResourceEnum cacheType;
+  private final NodeConfigCache nodeConfigCache;
+  private final double cacheSizeUpperBound;
+
+  private Long currentCacheMaxSizeInBytes;
+  private Long heapMaxSizeInBytes;
+  private long cacheUpperBoundInBytes;
+  private long desiredCacheMaxSizeInBytes;
+
+  private Map<ResourceEnum, Long> stepSizeInBytes = new HashMap<>();
+
+  public ModifyCacheMaxSizeAction(
+      final NodeKey esNode,
+      final ResourceEnum cacheType,
+      final NodeConfigCache nodeConfigCache,
+      final double cacheSizeUpperBound,
+      final boolean increase) {
+    // TODO: Add lower bound for caches
+    // TODO: Address cache scaling down  when JVM decider is available
+
+    this.esNode = esNode;
+    this.cacheType = cacheType;
+    this.nodeConfigCache = nodeConfigCache;
+    this.cacheSizeUpperBound = cacheSizeUpperBound;
+
+    setStepSize();
+
+    if (validateAndSetConfigValues()) {
+      long desiredCapacity =
+              increase ? currentCacheMaxSizeInBytes + getStepSize(cacheType) : currentCacheMaxSizeInBytes;
+      setDesiredCacheMaxSize(desiredCapacity);
+    }
+  }
+
+  @Override
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public boolean isActionable() {
+    if (currentCacheMaxSizeInBytes == null) {
+      return false;
+    }
+    return desiredCacheMaxSizeInBytes != currentCacheMaxSizeInBytes;
+  }
+
+  @Override
+  public long coolOffPeriodInMillis() {
+    return COOL_OFF_PERIOD_IN_MILLIS;
+  }
+
+  @Override
+  public List<NodeKey> impactedNodes() {
+    return Collections.singletonList(esNode);
+  }
+
+  @Override
+  public Map<NodeKey, ImpactVector> impact() {
+    final ImpactVector impactVector = new ImpactVector();
+    if (currentCacheMaxSizeInBytes != null) {
+      if (desiredCacheMaxSizeInBytes > currentCacheMaxSizeInBytes) {
+        impactVector.increasesPressure(HEAP);
+      } else if (desiredCacheMaxSizeInBytes < currentCacheMaxSizeInBytes) {
+        impactVector.decreasesPressure(HEAP);
+      }
+    }
+    return Collections.singletonMap(esNode, impactVector);
+  }
+
+  @Override
+  public String summary() {
+    if (!isActionable()) {
+      return String.format("No action to take for: [%s]", NAME);
+    }
+    return String.format(
+        "Update [%s] capacity from [%d] to [%d] on node [%s]",
+        cacheType.toString(),
+        currentCacheMaxSizeInBytes,
+        desiredCacheMaxSizeInBytes,
+        esNode.getNodeId());
+  }
+
+  @Override
+  public String toString() {
+    return summary();
+  }
+
+  public void setStepSizeForCache(final long stepSize) {
```

Comment:
Rename param to `stepSizeInBytes` to make callers aware of expected units.

Reply:
Done